data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateActionResponse.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import com.amazon.dataprepper.model.event.Event;
+
+import java.util.Optional;
+
+/**
+ * Model class to be returned in {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction}. Contains both the Event to be processed and an option
+ * to close the current window for {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateProcessor} immediately after an event is handled.
+ * @since 1.3
+ */
+public class AggregateActionResponse {
+    private Event event;
+
+    public AggregateActionResponse(final Event event) {
+        this.event = event;
+    }
+
+    public Optional<Event> getEvent() {
```

Comment:
Sorry for not being clear on my previous comment on the implication of having Optional in getter of POJO. If we try to serialize AggregateActionResponse object to JSON, we'll most likely see some unexpected data structure in the final JSON.

So, if we use Jackson to serialize AggregateActionResponse object, we'll get:
```
final ObjectMapper objectMapper = new ObjectMapper();
final Event event = constructEvent();
final AggregateActionResponse response = AggregateActionResponse.fromEvent(event);
System.out.println(objectMapper.writeValueAsString(response));
```

```
{"event":{"empty":false,"present":true}}
```

Consider returning Event for getEvent method and let the calling code convert this into Optional if needed.

```
public Event getEvent() { return event; }
```

Calling code

```
Optional<Event> event = Optional.ofNullable(aggregateActionResponse.getEvent());
```

Reply:
Is it an option to have 2 getters for the Event, one as the base Event and one as an Optional<Event>? I am pretty sure it is possible to tell Jackson which getter matters for serialization, and which one should be ignored.

Reply:
yeah, you can use JsonProperty annotation to a method specify how to deserialize the attribute.

```
@JsonProperty("event")
public Event getEvent() { return event; }

public Event getOptionalEvent() { return Optional.ofNullable(event); }

```

The only problem with approach is we are coupled with Jackson serialization/deserialization. If we decide to use some other form of serialization, we need to make sure to change this.

Is there any reason you want to have method return optional ? Can the calling code have the wrapper around this method ? I would not prefer it for POJO or model class, but for non-pojo class it should be good.

Reply:
The reason I would prefer returning an Optional is to avoid returning null in the function. When the getter is used and Event is returned as null it makes NPEs much more likely. 

However, I did some reading and you are correct that it isn't encouraged to return Optionals is getters like this. I do think that the your code snippet above where the calling code wraps in an Optional is good. I will make sure to document in the javadoc for `AggregateActionResponse` that Event can be null.

Reply:
I'd rather return only `Optional<Event>`. The field must be only `Event`. I believe Jackson will serialize this as an `Event` this way.

Reply:
I'm also fine with `Event getEvent()` if that is what we agree on. Most importantly, I'd like only one getter.